bson/unmarshal.go

```diff
@@ -31,7 +31,7 @@ type Unmarshaler interface {
 // document. To create custom BSON unmarshaling behavior for an entire BSON
 // document, implement the Unmarshaler interface instead.
 type ValueUnmarshaler interface {
-	UnmarshalBSONValue(Type, []byte) error
+	UnmarshalBSONValue(byte, []byte) error
```

Comment:
Optional: Consider adding named parameters to the interface to describe what the values are, since we're removing some type information. We may also want to update the documentation.

```suggestion
	UnmarshalBSONValue(typ byte, data []byte) error
```